src/main/java/com/github/alina/ui/Cli.java

```diff
@@ -62,4 +75,67 @@ public void registerUser() {
             default -> currentUser = adminService.register(name, userName, password);
         }
     }
+
+    enum UserType {
+        ADMIN,
+        TEACHER,
+        STUDENT,
+        NOTLOGIN
+
+    }
+
+    public void menu() {
+        switch (getUserType()) {
+            case NOTLOGIN -> {
+                System.out.println("You should login or register, 1 for login or 2 for register");
+                int option = scanner.nextInt();
+                if (option == 1) {
+                    login();
+                } else {
+                    registerUser();
+                }
+            }
+            case TEACHER -> {
+                System.out.println("1: List Courses, 2: List My Courses, 3: Grade Student");
+                int option = scanner.nextInt();
+                switch (option) {
+                    case 1 -> courseService.printAll();
+                    case 2 -> ((Teacher) currentUser).getTeachingCourses().forEach(System.out::println);
+                    case 3 -> {
+                        int studentId = scanner.nextInt();
+                        int courseId = scanner.nextInt();
+                        Student student = studentService.find(studentId);
+                        Course course = courseService.find(courseId);
+                        if (((Teacher) currentUser).getTeachingCourses().contains(course)) {
```

Comment:
casting `currentUser` to `Teacher` constantly can become problematic in the future. if you have to maybe cast it once earlier or create a new variable that's of type `Teacher` cast `currentUser` into it and use it throughout the function